diff --git a/third_party/fast_float/src/include/fast_float/float_common.h b/third_party/fast_float/src/include/fast_float/float_common.h
index 1265d9e..c826b95 100644
--- a/third_party/fast_float/src/include/fast_float/float_common.h
+++ b/third_party/fast_float/src/include/fast_float/float_common.h
@@ -7,11 +7,6 @@
 #include <cstring>
 #include <type_traits>
 #include <system_error>
-#ifdef __has_include
-#if __has_include(<stdfloat>) && (__cplusplus > 202002L || _MSVC_LANG > 202002L)
-#include <stdfloat>
-#endif
-#endif
 #include "constexpr_feature_detect.h"
 
 namespace fast_float {
